mlos_bench/mlos_bench/storage/base_storage.py

```diff
@@ -0,0 +1,274 @@
+                     tunables: TunableGroups, experiment_id: str, trial_id: int,
+                     config_id: int, opt_target: str, config: Optional[Dict[str, Any]] = None):
+            # pylint: disable=too-many-arguments
+            self._tunables = tunables
+            self._experiment_id = experiment_id
+            self._trial_id = trial_id
+            self._config_id = config_id
+            self._opt_target = opt_target
+            self._config = config or {}
+
+        def __repr__(self) -> str:
+            return f"{self._experiment_id}:{self._trial_id}"
+
+        @property
+        def trial_id(self) -> int:
+            """
+            ID of the current trial.
+            """
+            return self._trial_id
+
+        @property
+        def config_id(self) -> int:
+            """
+            ID of the current trial configuration.
+            """
+            return self._config_id
+
+        @property
+        def tunables(self) -> TunableGroups:
+            """
+            Tunable parameters of the current trial.
+            """
+            return self._tunables
+
+        def config(self, global_config: Optional[Dict[str, Any]] = None) -> dict:
+            """
+            Produce a copy of the global configuration updated
+            with the parameters of the current trial.
+            """
+            config = self._config.copy()
+            config.update(global_config or {})
+            config["experimentId"] = self._experiment_id
+            config["trialId"] = self._trial_id
+            return config
+
+        @abstractmethod
+        def update(self, status: Status,
+                   value: Optional[Union[Dict[str, Any], Any]] = None
+                   ) -> Optional[Dict[str, Any]]:
+            """
+            Update the storage with the results of the experiment.
+
+            Parameters
+            ----------
+            status : Status
+                Status of the experiment run.
+            value : Optional[Union[Dict[str, float], float]]
+                One or several metrics of the experiment run.
+                Must contain the optimization target if the status is SUCCEEDED.
+
+            Returns
+            -------
+            value : Optional[Dict[str, float]]
+                Same as value, but always in the dict format.
+            """
+            _LOG.info("Store trial: %s :: %s %s", self, status, value)
+            if isinstance(value, dict) and self._opt_target not in value:
+                _LOG.warning("Trial %s :: opt. target missing: %s", self, self._opt_target)
+                # raise ValueError(
+                #     f"Optimization target '{self._opt_target}' is missing from {value}")
+            return {self._opt_target: value} if isinstance(value, (float, int)) else value
+
+        @abstractmethod
+        def update_telemetry(self, status: Status,
+                             value: Optional[Dict[str, Any]] = None) -> None:
```

Comment:
same here: `Any` type hint, but docstring says `float`

Reply:
yep. `float` it is